src/core/egg-herbie.rkt

```diff
@@ -989,9 +991,76 @@
         ; expression of operators
         [(list (? operator-exists? op) ids ...) (cons op (map loop ids))])))
 
-  ; the actual extraction procedure
-  ; as long as the `id` is valid, extraction will work
-  (lambda (id) (cons (unsafe-eclass-cost id) (build-expr id))))
+  (list unsafe-eclass-cost build-expr))
+
+(define ((typed-egg-batch-extractor cost-proc batch-extract-to) regraph)
```

Comment:
Ok, I think this is basically a copy-and-paste of the normal typed egg extractor, except that it uses the batch stuff at the end, right? That's fine, I'm just making sure. Later we should clean up and merge them where possible

Reply:
Yes, that's right.
The main difference is that `typed-egg-batch-extractor` returns functions to extract nodes straight to the batch, while `typed-egg-extractor` returns function to extract nodes into expressions.
There are no other differences